### PR TITLE
push-hook: replace +should-proxy-update with +transform-proxy-update

### PR DESCRIPTION
**Pull request by tacryt-socryp**
This is a prerequisite for us to stop trusting client timestamps in `%graph-store`. Note that this retains all of the semantics and behavior of the `+should-proxy-update` arm, while adding the option of transforming the vase that the client supplies to the `%push-hook`.

Fixes: https://github.com/urbit/landscape/issues/270

Note: Needs close review and Ropsten testing before it is ready to go out. It is ready for review though!

### CODE REVIEW

**Comment by ixv** [comment]
I get this when trying to boot a fakezod: 
image

**Comment by tacryt-socryp** [comment]
Will boot up a new fakezod and try again.

**Comment by tacryt-socryp** [comment]
@ixv Resolved the issue. It was as simple as `+welp` instead of `+weld`.

**Comment by urcades** [comment]
stamp

**Comment by ixv** [comment]
ropsten net is up

File: pkg/arvo/lib/push-hook.hoon [hunk 1]
```diff
@@ -301,20 +301,20 @@
...
   ::
   ++  poke-update
-    |=  =vase
+    |=  vas=vase
     ^-  (quip card:agent:gall _state)
-    ?>  (should-proxy-update:og vase)
-    =/  wire
-      (make-wire /store)
+    =/  vax=(unit vase)  (transform-proxy-update:og vas)
+    ?>  ?=(^ vax)
```
Comment (liam-fitzgerald) [inline pkg/arvo/lib/push-hook.hoon:307]: `=/  =vase  (need (tranform-proxy-update:org vas))`

File: pkg/arvo/app/graph-push-hook.hoon [hunk 2]
```diff
@@ -66,28 +66,99 @@
...
-      %tags               %.n
-      %tag-queries        %.n
-      %run-updates        %.n
+      %add-nodes
+    ?.  (is-allowed-add:hc rid nodes.q.update)
+      ~
+    =/  mark  (get-mark:gra rid)
+    ?~  mark  `vas
+    |^
+    =/  transform=$-([index:store post:store atom ?] [index:store post:store])
```
Comment (liam-fitzgerald) [inline pkg/arvo/app/graph-push-hook.hoon:82]: This would be way cleaner as just a `=+` and `!<`

**Review by liam-fitzgerald — APPROVED** [review]
Couple style nits, nothing major. If this works well on testnet then LGTM!

**Comment by tacryt-socryp** [comment]
@ixv found an issue on the testnet, but I've pushed a fix for it here, and also applied it to `~tacryt`. It's unnecessary for it to be applied to the other ships, since `~tacryt` is the group host. Ready for more testing!

**Comment by tacryt-socryp** [comment]
If this is merged as is, it will regress the "pending message display" feature, so I'm putting this back to draft status and continuing a rewrite to the pending message system to use `node hash` as its unique identifier. As such, I'm building out a thread that we will send all `%add-nodes` events from the frontend to, which will return us a map from index to hash that we can use to generate "unique pending messages" on the frontend that get cleared as they currently do.

**Comment by tacryt-socryp** [comment]
@liam-fitzgerald Can you do a review pass on the %graph-add-nodes thread I just added? It compiles, but have yet to integrate it into the frontend.

**Comment by liam-fitzgerald** [comment]
Yeah `-graph-add-nodes` looks fine for that particular case

**Comment by tacryt-socryp** [comment]
Okay, as of now, this is ready for full-on testing, as it is feature-complete. @matildepark @ixv

**Comment by matildepark** [comment]
Will review the API and store logic but could you merge master back in? The store has a conflict, may affect the review.

**Comment by tacryt-socryp** [comment]
I've discovered issues today and been fixing them. Now, all of the backend actions work properly, but I still need to fix some interface display issues.

**Review by liam-fitzgerald — APPROVED** [review]

**Pull request merged**